test/python/circuit/library/test_pauli_feature_map.py

```diff
@@ -324,7 +324,7 @@ def test_pauli_zz_with_barriers(self):
 
         def zz(circuit, i, j):
             circuit.cx(i, j)
-            circuit.p(2 * (np.pi - params[i]) * (np.pi - params[j]), j)
+            circuit.p(2.0 * (params[j] - np.pi) * (params[i] - np.pi), j)
```

Comment:
Before this PR, the following evaluated to `True`:
```python
expr1 = (np.pi - x) * (np.pi - y)
expr2 = (-x + np.pi) * (-y + np.pi)
expr1 == expr2  # True before, now is False
```
I assume that is why this test needs changing. Changing this behavior seems a bit dangerous, since users might implicitly rely on the comparison here (and this would be very hard to track down).

Can we fix the above to also evaluate to `True`?

Reply:
my local installation returns True for this case.
I think comparing 2 circuits in the test case compares with string output from the equation, and it is hard to get the same string output by different implementation

Reply:
I will check all test cases I rewrote in this PR whether we can restore or not

Reply:
I'm sorry but this one can be restored as before, but for others are testing with string